src/zope/interface/interface.py

```diff
@@ -370,6 +370,10 @@ def __init__(self, name, bases=(), attrs=None, __doc__=None,
 
         self.__module__ = __module__
 
+        # local class definitions may share the same module and name,
+        # use qualname for hashes
+        self.__qualname__ = attrs.get('__qualname__', name)
```

Comment:
I'm concerned that there may be a lot of unexpected knock-on effects from defining this standard attribute now, mostly with tooling. But don't ask me exactly what they might be. 

What triggers that concern is that `__qualname__` is specifically rejected below, albeit for different reasons.

Reply:
A class should define a `__qualname__` if I get PEP 3155 right. Until now Interface subclasses did not. So I do not understand why 
>  there may be a lot of unexpected knock-on effects from defining this standard attribute now
May you add more details to your concern?

Reply:
> May you add more details to your concern?

I don't have specifics, but I'm thinking of things like this:
```python
if hasattr(cls, '__qualname__'):
    # We're running on Python 3. Do Python 3 things.
else:
    # We're on Python 2 *or* we're dealing with an interface. 
    # An interface is like, but not the same as, a real class
```

Suddenly code paths that weren't ever taken with an interface before could be. What happens depends on the code.